lib/core/presentation/pages/home/views/home_view.dart

```diff
@@ -0,0 +1,322 @@
+                              color: colorScheme.onPrimary,
+                              fontWeight: FontWeight.w600,
+                            ),
+                          ),
+                          SizedBox(height: Spacing.small),
+                          ...attendingEvents.map(
+                            (event) => Padding(
+                              padding: EdgeInsets.only(bottom: Spacing.xSmall),
+                              child: HomeEventCard(event: event),
+                            ),
+                          ),
+                          SizedBox(height: Spacing.medium),
+                        ],
+                      ]),
+                    );
+                  },
+                );
+              },
+            ),
+          ),
+        SliverPadding(
+          padding: EdgeInsets.symmetric(horizontal: Spacing.small),
+          sliver: SliverToBoxAdapter(
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Text(
+                  t.discover.discover.toUpperCase(),
+                  style: Typo.small.copyWith(
+                    color: colorScheme.onPrimary,
+                    fontWeight: FontWeight.w600,
+                  ),
+                ),
+                EventTimeFilterButton(
+                  onSelect: _selectEventTimeFilter,
+                ),
+              ],
+            ),
+          ),
+        ),
+        SliverToBoxAdapter(
+          child: SizedBox(
+            height: Spacing.superExtraSmall,
+          ),
+        ),
+        SliverPadding(
+          padding: EdgeInsets.symmetric(horizontal: Spacing.small),
+          sliver: SliverToBoxAdapter(
+            child: _EventList<HomeEventListingBloc>(
+              eventTimeFilter: eventTimeFilter,
+            ),
+          ),
+        ),
+        SliverToBoxAdapter(
+          child: SizedBox(
+            height: 120.h,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _EventList<T extends BaseEventListingBloc> extends StatelessWidget {
+  const _EventList({
+    super.key,
+    this.eventTimeFilter,
+  });
+  final EventTimeFilter? eventTimeFilter;
+
+  Widget _buildEmptyEvents(BuildContext context) {
+    final t = Translations.of(context);
+    final String timeFilterText =
+        eventTimeFilter != null ? t['common.${eventTimeFilter!.labelKey}'] : '';
+
+    return Center(
+      child: Text(
+        t.event.empty_home_events(time: timeFilterText),
+        textAlign: TextAlign.center,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<T, BaseEventsListingState>(
+      builder: (context, eventsListingState) {
+        return eventsListingState.when(
+          loading: () => Loading.defaultLoading(context),
+          fetched: (_, filteredEvents) {
+            if (filteredEvents.isEmpty) return _buildEmptyEvents(context);
+            return ListView.separated(
```

Comment:
You're using Sliver above, then why not directly use SliverList but ListView with shrinkWrap ? 
Should separate 3 specific widgets: HomeHostingEventsList, HomeMyEventsList, HomeDiscoverEventsList, 
The current code looks confusing